library/general/uniqueingresshost/src.rego

```diff
@@ -5,26 +5,12 @@ identical(obj, review) {
   obj.metadata.name == review.object.metadata.name
 }
 
-make_apiversion(kind) = apiVersion {
-  g := kind.group
-  v := kind.version
-  g != ""
-  apiVersion = sprintf("%v/%v", [g, v])
-}
-
-make_apiversion(kind) = apiVersion {
-  kind.group == ""
-  apiVersion = kind.version
-}
-
 violation[{"msg": msg}] {
   input.review.kind.kind == "Ingress"
-  apiVersion := make_apiversion(input.review.kind)
-  apis = ["extensions/v1beta1", "networking.k8s.io/v1beta1"]
-  apiVersion == apis[_]
+  re_match("^(extensions|networking.k8s.io)$", input.review.kind.group)
   host := input.review.object.spec.rules[_].host
-  other := data.inventory.namespace[ns][otherapi]["Ingress"][name]
-  otherapi == apis[_]
+  other := data.inventory.namespace[ns][otherapiversion]["Ingress"][name]
+  re_match("^(extensions|networking.k8s.io).+$", otherapiversion)
```

Comment:
add a `/` before the `.+$` to guarantee we aren't catching groups like: `extensions.something.org`